### PR TITLE
Clean up goby_launch directives

### LINKED ISSUE #42: goby_launch should have a way to tag commands that need to be cleaned up
From a discussion elsewhere:

Is it just going to like `killall socat`? Maybe it would be better for the launch script to be able to tag a command, like one of:

```
[kill=SIGTERM] socat --whatever
[not-goby] socat --whatever
```

Then the launch tool can know that it needs to grab the PID of the launched subprocess and kill it later.

### PR DESCRIPTION
**Pull request by tsaubergine**
Fixes #42

As suggested in #42, Instead of 

```
gobyd
socat ...
foo ...
# [goby_launch]: not_goby=socat,foo
```
we write
```
gobyd
[kill=SIGTERM] socat ...
[kill=SIGINT] foo ...
```
where `kill=` defaults to `kill=goby_terminate`, and you can use `kill=SIGINT`, `kill=SIGTERM`, `kill=goby_terminate`,  and `kill=SIGKILL`

### CODE REVIEW

**Comment by tsaubergine** [comment]
@rgov, I think this is exactly as suggested, please confirm that this works for you. Thanks for the feedback

File: scripts/goby_launch [hunk 1]
```diff
@@ -193,56 +170,88 @@ gobyd_interprocess=$(bash -c "$gobyd_escline --app 'debug_cfg: true'" | sed -n "
 gnterm=()
 while IFS= read -r line
 do
-    escline=$(echo $line | sed 's/"/\\"/g')
-    bin=$(echo $line | cut -d " " -f 1)
+    cmd=$(echo $line | sed 's/^ *\[.*\]//')
+    directive_str=
+    echo $line | egrep -q '\[.*\]'  &&
+        directive_str=$(echo $line | sed 's/^ *\[\(.*\)\].*$/\1/')
```
Comment (rgov) [inline scripts/goby_launch]: My suggestion is to use `\s` (if `sed` supports it) to allow for other whitespace in the indentation. The `.*$` is redundant.

File: scripts/goby_launch [hunk 2]
```diff
@@ -56,20 +56,16 @@ do_kill()
...
     # send SIGTERM to all the child processes
     for ((i=$((${#spid[@]}-1)); i>=0; --i)); do 
         p=${spid[i]}
         cmd=${spid_cmd[i]}
+        sig=${spid_sig[i]}
+
         kill -0 $p >& /dev/null || continue
 
-        if ! echo "$cmd" | egrep -q $not_goby_regex; then
+        if [ "$sig" == "goby_terminate" ]; then
```
Comment (rgov) [inline scripts/goby_launch]: `==` is non-standard; bash allows it, but it's supposed to be `=`

File: scripts/goby_launch [hunk 3]
```diff
@@ -157,28 +156,6 @@ if [ ! -e "$launchfile" ]; then
...
-            ;;
-        *)
-            echo "Invalid [goby_launch]: directive key: " $key
-            exit 1
-            ;;
-    esac                
-done < <(egrep "$launch_vars_regex" "$launchfile" | sed "s/$launch_vars_regex//")
 
 # search for gobyd so we can use the same interprocess { } configuration for goby_terminate
 gobyd_line=$(cat "$launchfile" | egrep "^ *[/a-zA-Z0-9_]*gobyd")
```
Comment (rgov) [inline scripts/goby_launch:161]: Not part of this commit but this part seems really fragile. Could you use a `[]` directive here too?

File: scripts/goby_launch (hunk 1 above)
Comment (rgov) [reply scripts/goby_launch]: Actually, `.*$` isn't redundant if you want to clear the rest of the line, sorry.

Comment (rgov) [reply scripts/goby_launch]: But you should quote `"$line"` when you're echoing it.
